scripts/mixins/families/chigoe.lua

```diff
@@ -5,21 +5,54 @@ require('scripts/globals/mixins')
 g_mixins = g_mixins or {}
 g_mixins.families = g_mixins.families or {}
 
-g_mixins.families.chigoe = function(chigoeMob)
-    chigoeMob:addListener('SPAWN', 'CHIGOE_SPAWN', function(mob)
+g_mixins.families.chigoe = function(chigoe)
+    chigoe:addListener('SPAWN', 'CHIGOE_SPAWN', function(mob)
         mob:hideName(true)
         mob:setUntargetable(true)
     end)
 
-    chigoeMob:addListener('ENGAGE', 'CHIGOE_ENGAGE', function(mob, target)
+    chigoe:addListener('ENGAGE', 'CHIGOE_ENGAGE', function(mob, target)
         mob:hideName(false)
         mob:setUntargetable(false)
     end)
 
-    chigoeMob:addListener('DISENGAGE', 'CHIGOE_DISENGAGE', function(mob, target)
+    chigoe:addListener('DISENGAGE', 'CHIGOE_DISENGAGE', function(mob, target)
         mob:hideName(true)
         mob:setUntargetable(true)
     end)
+
+    chigoe:addListener('CRITICAL_TAKE', 'CHIGOE_CRITICAL_TAKE', function(mob)
+        mob:setHP(0)
+    end)
+
+    chigoe:addListener('WEAPONSKILL_TAKE', 'CHIGOE_WEAPONSKILL_TAKE', function(target, wsid)
+        if wsid then
+            target:setHP(0)
+        end
+    end)
+
+    chigoe:addListener('ABILITY_TAKE', 'CHIGOE_ABILITY_TAKE', function(mob, user, ability)
+        local abilities =
+        {
+            46,  -- Shield Bash
+            66,  -- Jump
+            67,  -- High Jump
+            77,  -- Weapon Bash
+            82,  -- Chi Blast
+            150, -- Tomahawk
+            170, -- Angon
+            201, -- Quickstep
+            202, -- Boxstep
+            203, -- Stutter Step
+            312, -- Feather Step
```

Comment:
can the script use `require('scripts/globals/mixins')` and then refer to it using xi.xi.jobAbility.Name? (dunno if it can, just thought it would be cool to avoid magic numbers)

Reply:
should be good, moved this into a set.